data/src/main/java/com/linkedin/data/codec/AbstractJacksonDataCodec.java

```diff
@@ -51,13 +51,26 @@
  */
 public abstract class AbstractJacksonDataCodec implements DataCodec
 {
+  /**
+   * Default factory to be shared between all jackson JSON codec instances. This is done to maximize factory reuse for
+   * performance reasons as recommended by Jackson authors.
+   *
+   * <a href="https://github.com/FasterXML/jackson-docs/wiki/Presentation:-Jackson-Performance">Jackson Performance</a>
+   */
+  public static final JsonFactory JSON_FACTORY = new JsonFactory();
+  static {
+    JSON_FACTORY.disable(JsonFactory.Feature.INTERN_FIELD_NAMES);
+  }
+
   protected static final int DEFAULT_BUFFER_SIZE = 4096;
 
   protected final JsonFactory _factory;
 
   protected AbstractJacksonDataCodec(JsonFactory factory)
   {
     _factory = factory;
+    // Disable string interning since it causes GC issues.
+    _factory.disable(JsonFactory.Feature.INTERN_FIELD_NAMES);
```

Comment:
This means it wouldn't be possible to override this behavior to enable interning

Reply:
You can. You can always inject a custom factory and in your constructor enable interning by explicitly invoking it after calling super.

Reply:
It should be possible to customize the jsonfactory with the existing codec classes, without having to create a subclass.

For example, it would help some service to override the default behavior. 
Since we are disabling interning when the default constructor is used, it should cover the default behavior. When a custom json factory is provided, we should treat it as override and not force behavior on it.